files/default/neutron-ha-tool.py

```diff
@@ -963,6 +963,27 @@ def pick(self):
         return self.agents_by_id[agent_id]
 
 
+class SingleAgentPicker(object):
+    agent_keys = ('id', 'host')
+    agent_selection_value = None
+
+    def __init__(self, qclient, agents):
+        self.agents = agents
+
+    def find_agent(self, value):
```

Comment:
`value` is a very vague variable name.  Please give it a name which reflects what the value actually represents.  IIUC, it actually represents `agent_id_or_host_id`, and IMHO the ugliness of that name is a code smell which suggests it's not ideal to overload a single method with two different types of search.  For example there is perhaps a chance (however astronomically small) that a given agent id is also a valid host id.  Perhaps a more likely issue would be that someone in the future writes some buggy code which accidentally passes an agent id when they actually meant to pass a host id, or vice-versa.  This overloading would hide that bug.  Or even more likely, someone manually passes the wrong id via the CLI.

I'd prefer to have separate code paths, e.g. `find_agent_by_id` and `find_agent_by_host_id`, or at least `def find_agent(self, object_type, object_id)`.  This would also make it clearer how often each of the code paths get used.  If one never gets used then it would be better to drop it.